spec/models/mentor_management_spec.rb

```diff
@@ -0,0 +1,123 @@
+      MentorManagement.assign_mentor(assignment.id, team.id)
+      expect(Team).to receive(:add_member).exactly(0).times
+    end
+
+    it 'returns early if team already has a mentor' do
+      allow(Assignment).to receive(:find).with(assignment.id).and_return(assignment)
+      allow(Team).to receive(:find).with(team.id).and_return(team)
+      # stub the call to `team.participants` so that `any?` returns `true`
+      allow(team).to receive(:participants).and_return([mentor])
+      MentorManagement.assign_mentor(assignment.id, team.id)
+      expect(Team).to receive(:add_member).exactly(0).times
+    end
+
+    it 'assigns a mentor to a team when the team size passes 50% max capacity' do
+      allow(Assignment).to receive(:find).with(assignment.id).and_return(assignment)
+      allow(Team).to receive(:find).with(team.id).and_return(team)
+
+      # add 2 students to our team
+      [student1, student2].each { |student| FactoryBot.create(:team_user, team_id: team.id, user_id: student.id)}
+
+      allow(assignment).to receive(:topics?).and_return(false)
+      allow(team).to receive(:topics).and_return(nil)
+
+      allow(MentorManagement).to receive(:select_mentor).with(assignment.id).and_return(ta)
+      allow(team).to receive(:add_member).and_return(true)
+
+      # if we've made it this far without failing, then there's nothing
+      # left for Mentor Management to test. The only question left to answer
+      # is "does Team#add_member work as expected?," and that is tested elsewhere.
+      # From The Magic Tricks of Testing Video (Week 9), it is not this class's job
+      # to test whether Team works properly.
+      MentorManagement.assign_mentor(assignment.id, team.id)
+    end
+  end
+
+  describe '#user_a_mentor?' do
+    it 'should return true if user is a mentor' do
+      expect(MentorManagement.user_a_mentor?(student1)).to be false
+      expect(MentorManagement.user_a_mentor?(ta)).to be true
+    end
+  end
+
+  describe '#get_mentors_for_assignment' do
+    it 'returns all mentors for the given assignment' do
+      allow(Participant).to receive(:where)
+                              .with(parent_id: assignment.id, duty: Participant::DUTY_MENTOR)
+                              .and_return([mentor])
+      mentor_user = MentorManagement.mentors_for_assignment(assignment.id).first
+      expect(mentor_user).to eq mentor
+    end
+  end
+
+  describe '#zip_mentors_with_team_count' do
+    it 'returns sorted tuples of (mentor ID, # of teams they mentor)' do
+      team_count = 3
+      r = Random.new(42)
+      team_ids = team_count.times.map {
+        random_id = r.rand(1000..10000)
+        FactoryBot.create(:team, id: random_id)
+        random_id
+      }
+      team_ids.each { |team_id| FactoryBot.create(:team_user, team_id: team_id, user_id: ta.id) }
+      expect(MentorManagement.zip_mentors_with_team_count(assignment.id)).to eq [[mentor.user_id, team_count]]
+    end
```

Comment:
can you please add a test to make sure your method that sends out the email successfully interfaces with the mailer

Reply:
We determined such a test is unnecessary since it tests nothing about MentorManagement, just that ActiveMailer works as expected.

Reply:
that's fair! we have tests for ActiveMailer elsewhere so this should be okay! Thanks,